test/includes/lxd.sh

```diff
@@ -1,5 +1,71 @@
 # LXD-related test helpers.
 
+spawn_lxd_snap() {
+    { set +x; } 2>/dev/null
+    # Install the snap and sideload the binaries
+    if ! [ -e /var/snap/lxd/common/lxd.debug ]; then
+      sideload_lxd_snap
+    fi
+
+    # Ensure LXD_DIR always points to the snap common dir
+    LXD_DIR_NON_SNAP="${LXD_DIR}"
+    LXD_DIR="/var/snap/lxd/common/lxd"
+
+    # For snap based tests, the lxc and lxc_remote functions MUST NOT be used
+    unset -f lxc lxc_remote
+
+    # shellcheck disable=SC2153
+    local lxd_backend="${LXD_BACKEND}"
+    if [ "$LXD_BACKEND" = "random" ]; then
+        lxd_backend="$(random_storage_backend)"
+    fi
+
+    if [ "${lxd_backend}" = "ceph" ] && [ -z "${LXD_CEPH_CLUSTER:-}" ]; then
+        echo "A cluster name must be specified when using the CEPH driver." >&2
```

Comment:
why is this? You can use ceph with a non-clustered lxd?

Reply:
It is referring to the `ceph` cluster name to create OSDs in. It's also included in the regular non-snap function which is why I included it there as well.